tests/testthat/test-compute_metrics.R

```diff
@@ -0,0 +1,300 @@
+      bootstraps(mtcars, 5),
+      grid = 3,
+      metrics = m_set_rsq,
+      control = control_grid(save_pred = TRUE)
+    )
+
+  collected_sum_rsq <- collect_metrics(res_rsq)
+  computed_sum_rsq  <- compute_metrics(res_rmse, m_set_rsq)
+
+  expect_equal(collected_sum_rsq, computed_sum_rsq)
+
+  collected_sum_rmse <- collect_metrics(res_rmse)
+  computed_sum_rmse  <- compute_metrics(res_rsq, m_set_rmse)
+
+  expect_equal(collected_sum_rsq, computed_sum_rsq)
+
+  collected_unsum_rsq <- collect_metrics(res_rsq, summarize = FALSE)
+  computed_unsum_rsq  <- compute_metrics(res_rmse, m_set_rsq, summarize = FALSE)
+
+  expect_equal(collected_unsum_rsq, computed_unsum_rsq)
+
+  collected_unsum_rmse <- collect_metrics(res_rmse, summarize = FALSE)
+  computed_unsum_rmse  <- compute_metrics(res_rsq, m_set_rmse, summarize = FALSE)
+
+  expect_equal(collected_unsum_rmse, computed_unsum_rmse)
+})
+
+test_that("sensitivity to `metrics` argument (compatible class metric types)", {
+  skip_on_cran()
+  skip_if_not_installed("kknn")
+
+  library(parsnip)
+  library(rsample)
+  library(yardstick)
+  library(modeldata)
+
+  set.seed(1)
+
+  res_acc_auc <-
+    tune_grid(
+      nearest_neighbor("classification", neighbors = tune()),
+      Class ~ .,
+      vfold_cv(two_class_dat, 5),
+      metrics = metric_set(accuracy, roc_auc),
+      grid = 3,
+      control = control_grid(save_pred = TRUE)
+    )
+
+  set.seed(1)
+
+  res_class <-
+    tune_grid(
+      nearest_neighbor("classification", neighbors = tune()),
+      Class ~ .,
+      vfold_cv(two_class_dat, 5),
+      metrics = metric_set(precision),
+      grid = 3,
+      control = control_grid(save_pred = TRUE)
+    )
+
+  set.seed(1)
+
+  res_prob <-
+    tune_grid(
+      nearest_neighbor("classification", neighbors = tune()),
+      Class ~ .,
+      vfold_cv(two_class_dat, 5),
+      metrics = metric_set(mn_log_loss),
+      grid = 3,
+      control = control_grid(save_pred = TRUE)
+    )
+
+  collected_sum_class <- collect_metrics(res_class)
+  computed_sum_class  <- compute_metrics(res_acc_auc, metric_set(precision))
+
+  expect_equal(collected_sum_class, computed_sum_class)
+
+  collected_unsum_class <- collect_metrics(res_class, summarize = FALSE)
+  computed_unsum_class  <- compute_metrics(res_acc_auc, metric_set(precision),
+                                           summarize = FALSE)
+
+  expect_equal(collected_unsum_class, computed_unsum_class)
+
+  collected_sum_prob <- collect_metrics(res_prob)
+  computed_sum_prob  <- compute_metrics(res_acc_auc, metric_set(mn_log_loss))
+
+  expect_equal(collected_sum_prob, computed_sum_prob)
+
+  collected_unsum_prob <- collect_metrics(res_prob, summarize = FALSE)
+  computed_unsum_prob  <- compute_metrics(res_acc_auc, metric_set(mn_log_loss),
+                                        summarize = FALSE)
+
+  expect_equal(collected_unsum_prob, computed_unsum_prob)
+})
+
+test_that("sensitivity to `metrics` argument (differing class metric types)", {
```

Comment:
I thought that this title meant that we were using the `sensitivity()` function for a second.